### PR TITLE
Feat: Shipping Simulation API

### PR DESCRIPTION
**Pull request by eduardoformiga**
## What's the purpose of this pull request?

This PR aims to Integrate the `ShippingSimulation` feature API.

Depends on https://github.com/vtex/faststore/pull/1440.

## How does it work?

Now, the `@faststore/api` is able to bring shipping simulation data and a location, giving some inputs, using the `useShippingSimulation` hook. And finally, we can remove the mock.

Sometimes the API that returns the address didn't bring the data we need like the neighborhood.  Thus, some treats were made to avoid showing null data.

|Figma Prototype | PDP Directly| PDP Using Regionalization| 
|-|-|-|
|Screen Shot 2022-08-22 at 17 49 19|fQAC5sAvSz|35G5o3GOvM|

## How to test it?

The account/product could not have the Shipping configs set. So, you can play around with some products or switch the account to another like `storecomponents` [here](https://github.com/vtex-sites/nextjs.store/blob/8573e234dd32cfa00c64dbb51b4612b78f2d5e57/store.config.js#L17) for example, and play with the `classic-shoes-37/p` product.

Screen Shot 2022-10-20 at 22 43 43

## References
ShippingSimulation Component https://github.com/vtex-sites/nextjs.store/pull/183
Figma https://www.figma.com/file/EYJqnbWByrSX9EMKvBzBLm/22%2F06%2F14-Handoff-for-PDP-Components?node-id=1001%3A131765
@faststore/api https://github.com/vtex/faststore/pull/1440
https://github.com/vtex/faststore/pull/1018
https://github.com/vtex-apps/shipping-estimate-translator

### CODE REVIEW

File: src/components/ui/ShippingSimulation/useShippingSimulation.ts [hunk 1]
```diff
@@ -12,7 +16,7 @@ type InputProps = {
 type ShippingOptionProps = {
   carrier: string
   estimate: string
-  price: number
+  price?: string
```
Comment (filipewl) [inline src/components/ui/ShippingSimulation/useShippingSimulation.ts:19]: Why did this change type?

File: src/components/ui/ShippingSimulation/useShippingSimulation.ts [hunk 2]
```diff
@@ -134,8 +109,28 @@ const reducer = (state: State, action: Action) => {
...
+  shipping: ShippingSimulationQueryQuery['shipping']
+): [string, ShippingOptionProps[]] {
+  const location =
+    [shipping?.address?.neighborhood, shipping?.address?.city]
+      .filter(Boolean)
+      .join(' / ') ?? ''
+
+  const options: ShippingOptionProps[] =
+    shipping?.logisticsInfo?.[0]?.slas?.map((sla) => ({
+      carrier: sla?.name ?? '',
```
Comment (filipewl) [inline src/components/ui/ShippingSimulation/useShippingSimulation.ts:122]: What do you think of favoring `friendlyName`? I haven't been able to query something that would return this field, but I imagine it's a more human-friendly version of `name`, when it's available.

CleanShot 2022-08-25 at 17 26 39@2x

```suggestion
      carrier: sla?.friendlyName ?? sla?.name ?? '',
```

**Review by filipewl — COMMENTED** [review]
Looking good Forms! Before approving I left some comments.

File: package.json [hunk 3]
```diff
@@ -31,7 +31,7 @@
     "@envelop/graphql-jit": "^1.1.1",
     "@envelop/parser-cache": "^2.2.0",
     "@envelop/validation-cache": "^2.2.0",
-    "@faststore/api": "^1.12.16",
+    "@faststore/api": "https://pkg.csb.dev/vtex/faststore/commit/bd2edcb4/@faststore/api",
```
Comment (lucasfp13) [inline package.json:34]: Just a reminder to update here after faststore PR is merged.

**Review by lucasfp13 — APPROVED** [review]
LGTM, great work!

File: src/components/ui/ShippingSimulation/ShippingSimulation.tsx [hunk 4]
```diff
@@ -80,21 +86,23 @@ function ShippingSimulation({
             <TableBody>
               {shippingOptions.map((option) => (
                 <TableRow
-                  key={option.carrier}
+                  key={option.friendlyName}
```
Comment (victorhmp) [inline src/components/ui/ShippingSimulation/ShippingSimulation.tsx:89]: https://github.com/vtex/faststore/pull/1440/files#r1017011607

File: src/components/ui/ShippingSimulation/ShippingSimulation.tsx [hunk 5]
```diff
@@ -80,21 +86,23 @@ function ShippingSimulation({
...
                   data-fs-shipping-simulation-table-row
                 >
                   <TableCell data-fs-shipping-simulation-table-cell>
-                    {option.carrier}
+                    {option.friendlyName}
                   </TableCell>
                   <TableCell data-fs-shipping-simulation-table-cell>
-                    {option.estimate}
+                    {option.friendlyShippingEstimate}
```
Comment (victorhmp) [inline src/components/ui/ShippingSimulation/ShippingSimulation.tsx:96]: https://github.com/vtex/faststore/pull/1440/files#r1017003538

**Pull request merged**